apps/web/middleware.ts

```diff
@@ -63,18 +79,8 @@ const routingForms = {
 };
 
 export const config = {
-  // Next.js Doesn't support spread operator in config matcher, so, we must list all paths explicitly here.
-  // https://github.com/vercel/next.js/discussions/42458
-  matcher: [
-    "/:path*/embed",
-    "/api/trpc/:path*",
-    "/login",
-    "/auth/login",
-    /**
-     * Paths required by routingForms.handle
-     */
-    "/apps/routing_forms/:path*",
```

Comment:
Are these removals intended? tagging @hariombalhara who might have more context on this.

Reply:
This change was required to ensure that middleware is executed on each page request. This is needed to set the `x-url` and `x-csp` headers that are used in RootLayout. With previous matcher settings we executed middleware only on some specific routes, which resulted in a bug where some app dir pages missed `x-url` header. 
We are open to  discuss alternative solutions. 
Also, we can add app dir pages paths to the matcher one by one as we migrate pages. So for current PR matcher will remain the same.